terraform/constellation-cluster/main.tf

```diff
@@ -34,16 +34,48 @@ resource "terraform_data" "config_generate" {
   ]
 }
 
-
-resource "null_resource" "config" {
+resource "null_resource" "aws_config" {
+  count = var.aws_config != null ? 1 : 0
   provisioner "local-exec" {
     command = <<EOT
-      if [ "${var.csp}" = "aws" ]; then
       ./yq eval '.provider.aws.region = "${var.aws_config.region}"' -i constellation-conf.yaml
       ./yq eval '.provider.aws.zone = "${var.aws_config.zone}"' -i constellation-conf.yaml
       ./yq eval '.provider.aws.iamProfileControlPlane = "${var.aws_config.iam_instance_profile_control_plane}"' -i constellation-conf.yaml
       ./yq eval '.provider.aws.iamProfileWorkerNodes = "${var.aws_config.iam_instance_profile_worker_nodes}"' -i constellation-conf.yaml
-      fi
+    EOT
+  }
+  triggers = {
+    always_run = timestamp()
+  }
+  depends_on = [
+    terraform_data.config_generate
+  ]
+}
+
+resource "null_resource" "gcp_config" {
+  count = var.gcp_config != null ? 1 : 0
+  provisioner "local-exec" {
+    command = <<EOT
+      ./yq eval '.provider.gcp.project = "${var.gcp_config.project}"' -i constellation-conf.yaml
+      ./yq eval '.provider.gcp.region = "${var.gcp_config.region}"' -i constellation-conf.yaml
+      ./yq eval '.provider.gcp.zone = "${var.gcp_config.zone}"' -i constellation-conf.yaml
+      ./yq eval '.provider.gcp.serviceAccountKeyPath = "${var.gcp_config.serviceAccountKeyPath}"' -i constellation-conf.yaml
+
```

Comment:
```suggestion
```

Reply:
I thought it was nice to seperate conf and state configuration